target/snitch_cluster/sw/apps/snax-mac-mult/src/snax-mac-mult.c

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//----------------------------------
+// Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+// Description:
+// This test checks if a single Snitch core
+// Can control multiple accelerators attached to it.
+//
+// Note that this test encodes the CSRs in a manual way
+// because the write_csr and read_csr functions
+// found in sw/deps/riscv-opcodes/encoding.h
+// Do not allow variable inputs. The values need to be
+// explicitly stated due to the # stringitized arguments
+//----------------------------------
+
+#include "snrt.h"
+
+#include "data.h"
+#include "mac.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    uint32_t *local_a_acc_0, *local_b_acc_0, *local_o_acc_0;
+    uint32_t *local_a_acc_1, *local_b_acc_1, *local_o_acc_1;
+    uint32_t *local_a_acc_2, *local_b_acc_2, *local_o_acc_2;
+    uint32_t *local_a_acc_3, *local_b_acc_3, *local_o_acc_3;
+
+    uint32_t SLICE_LEN = VEC_LEN / ACC_NUM;
+
+    // Calculate base addresses
+    local_a_acc_0 = (uint32_t *)snrt_l1_next();
+    local_b_acc_0 = local_a_acc_0 + VEC_LEN;
+    local_o_acc_0 = local_b_acc_0 + VEC_LEN;
+
+    local_a_acc_1 = local_a_acc_0 + SLICE_LEN;
+    local_b_acc_1 = local_b_acc_0 + SLICE_LEN;
+    local_o_acc_1 = local_o_acc_0 + SLICE_LEN;
+
+    local_a_acc_2 = local_a_acc_0 + 2 * SLICE_LEN;
+    local_b_acc_2 = local_b_acc_0 + 2 * SLICE_LEN;
+    local_o_acc_2 = local_o_acc_0 + 2 * SLICE_LEN;
+
+    local_a_acc_3 = local_a_acc_0 + 3 * SLICE_LEN;
+    local_b_acc_3 = local_b_acc_0 + 3 * SLICE_LEN;
+    local_o_acc_3 = local_o_acc_0 + 3 * SLICE_LEN;
+
+    // Manual declaration to optimize the part
+    // Where the compiler has to fetch the VEC_LEN
+    // From memory if we plug it directly into the
+    // csr_write functions. Adding this line
+    // pre-computes and does not need to do
+    // unnecessary lw or li instructions during csr setup
```

Comment:
is this supposed to go right about to SLICE_LEN?
Don't worry about this not being constant folded. This is what the acc dialect is for in SNAX-MLIR

Reply:
I took the comment out haha there was a separate variable doing this. I just noticed in the traces that it computes inside. So I figured to put some other variables outside of the `if snrt_is_compute()` condition instead to avoid having to compute while during setup.